### PR TITLE
Kavitha, Umair | A-1204370983916597| add database column and api changes for priority

### PR DESCRIPTION
**Pull request by kavitha-sundararajan**
New column in database for appointment priority. 
API able to save and return the priority

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/appointments/model/AppointmentPriority.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.openmrs.module.appointments.model;
+
+public enum AppointmentPriority {
+    AsNeeded("AsNeeded"), Routine("Routine"), Emergency("Emergency"), Invalid("Invalid");
```
Comment (binduak) [inline api/src/main/java/org/openmrs/module/appointments/model/AppointmentPriority.java:4]: @kavitha-sundararajan confirming these are the FHIR priority status ? I remember there is "stat" as status

Comment (kavitha-sundararajan) [reply api/src/main/java/org/openmrs/module/appointments/model/AppointmentPriority.java:4]: Yes @binduak , Akhil has gone through the different types and suggested use these three "AsNeeded, Routing and Emergency" and we added "Invalid" for identifying invalid types coming from the API

Comment (binduak) [reply api/src/main/java/org/openmrs/module/appointments/model/AppointmentPriority.java:4]: ok..these are FHIR status ?

Comment (kavitha-sundararajan) [reply api/src/main/java/org/openmrs/module/appointments/model/AppointmentPriority.java:4]: Others except "Invalid" are FHIR status [https://terminology.hl7.org/5.1.0/ValueSet-v3-ActPriority.html]. Is it okay to have that ?

Comment (binduak) [reply api/src/main/java/org/openmrs/module/appointments/model/AppointmentPriority.java:4]: yes @kavitha-sundararajan

**Comment by binduak** [comment]
@kavitha-sundararajan will raise a new PR to remove **Invalid** appointment priority type and throw exception if invalid priority has sent from UI/API

**Pull request merged**